pkg/operator/controller/controller_dns.go

```diff
@@ -1,89 +1,135 @@
 package controller
 
 import (
+	"context"
 	"fmt"
 
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/manifests"
+
 	operatorv1 "github.com/openshift/api/operator/v1"
-	"github.com/openshift/cluster-ingress-operator/pkg/dns"
 
 	corev1 "k8s.io/api/core/v1"
 
-	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ensureDNS will create DNS records for the given LB service. If service is
 // nil, nothing is done.
-func (r *reconciler) ensureDNS(ci *operatorv1.IngressController, service *corev1.Service, dnsConfig *configv1.DNS) error {
-	records := desiredDNSRecords(ci, dnsConfig, service)
-	for _, record := range records {
-		err := r.DNSManager.Ensure(record)
-		if err != nil {
-			return fmt.Errorf("failed to ensure DNS record %v for %s/%s: %v", record, ci.Namespace, ci.Name, err)
-		}
-		log.Info("ensured DNS record for ingresscontroller", "namespace", ci.Namespace, "name", ci.Name, "record", record)
+func (r *reconciler) ensureWildcardDNSRecord(ic *operatorv1.IngressController, service *corev1.Service) (*iov1.DNSRecord, error) {
+	if service == nil {
+		return nil, nil
 	}
-	return nil
-}
 
-func newAliasRecord(domain, target string, zone configv1.DNSZone) *dns.Record {
-	return &dns.Record{
-		Zone: zone,
-		Type: dns.ALIASRecord,
-		Alias: &dns.AliasRecord{
-			Domain: domain,
-			Target: target,
-		},
+	desired := desiredWildcardRecord(ic, service)
```

Comment:
Could `service.Status.LoadBalancer.Ingress` change from non-empty to empty, and should we handle that case?  Probably that would only happen in case of an intermittent outage, in which case it leaving the old record might be preferable.  In any case, I'd like us to make a deliberate choice, and document it.

Reply:
Added a TODO with these considerations. Not going to be able to solve it here yet.